### PR TITLE
fix(Java): Improve Collection of Errors string

### PR DESCRIPTION
**Pull request by texastony**
_Issue #, if available:_

_Description of changes:_

Collection of Errors is a frustrating error because it requires customer action
completely serialize it into logs.

We can make a simple fix in Java to improve the CX,
by always serializing the list into the Collection Of Error's
message.

We concatenate the given error message with 
a serialization of all the nested errors.

_Squash/merge commit message, if applicable:_
```
fix(Java): Improve Collection of Errors string
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AwsCryptographyPrimitives/codegen-patches/java/dafny-4.9.0.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ public class CollectionOfErrors extends RuntimeException {
+ 
+@@ -134,6 +135,14 @@ public class CollectionOfErrors extends RuntimeException {
+     }
+ 
+     public CollectionOfErrors build() {
++      if (!(this.list == null || this.list.isEmpty())) {
++        this.message =
++          this.message +
++          " String rep of Exceptions in list.\n" +
```
Comment (rishav-karanjit) [inline AwsCryptographyPrimitives/codegen-patches/java/dafny-4.9.0.patch:20]:
```suggestion
+          " String representation of Exceptions in list.\n" +
```

Comment (rishav-karanjit) [reply AwsCryptographyPrimitives/codegen-patches/java/dafny-4.9.0.patch:20]: And all other places

**Review by rishav-karanjit — CHANGES_REQUESTED** [review]

File: AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographicMaterialProviders/java/dafny-4.9.0.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+index 796adaa3e..d1eea14c4 100644
+--- b/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/materialproviders/model/CollectionOfErrors.java
++++ a/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/materialproviders/model/CollectionOfErrors.java
+@@ -4,6 +4,9 @@
+ package software.amazon.cryptography.materialproviders.model;
+ 
+ import java.util.List;
++import java.util.stream.Collectors;
++import software.amazon.awssdk.utils.CollectionUtils;
++import software.amazon.awssdk.utils.StringUtils;
```
Comment (rishav-karanjit) [inline AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographicMaterialProviders/java/dafny-4.9.0.patch:11]: `CollectionUtils` and `StringUtils` seems to be unused.

Comment (rishav-karanjit) [reply AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographicMaterialProviders/java/dafny-4.9.0.patch:11]: Sorry. Missed this in my 1st review.

File: AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographyKeyStore/java/dafny-4.9.0.patch [hunk 3]
```diff
@@ -0,0 +1,28 @@
+diff --git b/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystore/model/CollectionOfErrors.java a/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystore/model/CollectionOfErrors.java
+index 9247dd55f..e74f3e03a 100644
+--- b/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystore/model/CollectionOfErrors.java
++++ a/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystore/model/CollectionOfErrors.java
+@@ -4,6 +4,8 @@
+ package software.amazon.cryptography.keystore.model;
+ 
+ import java.util.List;
++import java.util.stream.Collectors;
++import software.amazon.awssdk.utils.CollectionUtils;
```
Comment (rishav-karanjit) [inline AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographyKeyStore/java/dafny-4.9.0.patch:10]: `CollectionUtils` and `StringUtils` seems to be unused.

Comment (rishav-karanjit) [reply AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographyKeyStore/java/dafny-4.9.0.patch:10]: This and other places

File: AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographicMaterialProviders/java/dafny-4.9.0.patch (hunk 2 above)
Comment (texastony) [reply AwsCryptographicMaterialProviders/codegen-patches/AwsCryptographicMaterialProviders/java/dafny-4.9.0.patch:11]: Hmm... that must have been from an earlier attempt on my part.
All is groovy.
I put my revision script in the last commit message!

**Pull request merged**